commit 7ce5716e13cfb37a86c02fe158403c002eb1b504
Author: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
Date:   Tue Nov 28 17:17:18 2023 -0800

    drm/xe/huc: Prepare for 2-step HuC authentication
    
    Starting on MTL, the HuC is authenticated twice, once via GuC (same as
    with older integrated platforms) and once via GSC; the first
    authentication allows the HuC to be used for clear-media workloads,
    while the second one unlocks support for protected content.
    Ahead of adding the authentication flow via GSC, this patch adds support
    for differentiating the 2 auth steps and checking if they're complete.
    
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Cc: Alan Previn <alan.previn.teres.alexis@intel.com>
    Cc: John Harrison <John.C.Harrison@Intel.com>
    Reviewed-by: Vivaik Balasubrawmanian <vivaik.balasubrawmanian@intel.com>
    Signed-off-by: Rodrigo Vivi <rodrigo.vivi@intel.com>

diff --git a/drivers/gpu/drm/xe/xe_uc.c b/drivers/gpu/drm/xe/xe_uc.c
index 72a7b3c2577d..25e1ddfd2f86 100644
--- a/drivers/gpu/drm/xe/xe_uc.c
+++ b/drivers/gpu/drm/xe/xe_uc.c
@@ -176,7 +176,7 @@ int xe_uc_init_hw(struct xe_uc *uc)
 		return ret;
 
 	/* We don't fail the driver load if HuC fails to auth, but let's warn */
-	ret = xe_huc_auth(&uc->huc);
+	ret = xe_huc_auth(&uc->huc, XE_HUC_AUTH_VIA_GUC);
 	xe_gt_assert(uc_to_gt(uc), !ret);
 
 	/* GSC load is async */